query-engine/js-connectors/src/proxy.rs

```diff
@@ -82,21 +87,75 @@ impl JSResultSet {
 #[napi]
 #[derive(Debug)]
 pub enum ColumnType {
+    // [PLANETSCALE_TYPE] (MYSQL_TYPE) -> [TypeScript example]
+    /// The following PlanetScale type IDs are mapped into Int32:
+    /// - INT8 (TINYINT) -> e.g. `127`
+    /// - INT16 (SMALLINT) -> e.g. `32767`
+    /// - INT24 (MEDIUMINT) -> e.g. `8388607`
+    /// - INT32 (INT) -> e.g. `2147483647`
     Int32,
+
+    /// The following PlanetScale type IDs are mapped into Int64:
+    /// - INT64 (BIGINT) -> e.g. `"9223372036854775807"` (String-encoded)
     Int64,
+
+    /// The following PlanetScale type IDs are mapped into Float:
+    /// - FLOAT32 (FLOAT) -> e.g. `3.402823466`
     Float,
+
+    /// The following PlanetScale type IDs are mapped into Double:
+    /// - FLOAT64 (DOUBLE) -> e.g. `1.7976931348623157`
     Double,
-    Text,
-    Enum,
-    Bytes,
+
+    /// The following PlanetScale type IDs are mapped into Numeric:
+    /// - DECIMAL (DECIMAL) -> e.g. `"99999999.99"` (String-encoded)
+    Numeric,
+
+    /// The following PlanetScale type IDs are mapped into Boolean:
+    /// - BOOLEAN (BOOLEAN) -> e.g. `1`
     Boolean,
+
+    /// The following PlanetScale type IDs are mapped into Char:
+    /// - CHAR (CHAR) -> e.g. `"c"` (String-encoded)
     Char,
-    Array,
-    Numeric,
-    Json,
-    DateTime,
+
+    /// The following PlanetScale type IDs are mapped into Text:
+    /// - TEXT (TEXT) -> e.g. `"foo"` (String-encoded)
+    /// - VARCHAR (VARCHAR) -> e.g. `"foo"` (String-encoded)
+    Text,
+
+    /// The following PlanetScale type IDs are mapped into Date:
+    /// - DATE (DATE) -> e.g. `"2023-01-01"` (String-encoded, yyyy-MM-dd)
     Date,
+
+    /// The following PlanetScale type IDs are mapped into Time:
+    /// - TIME (TIME) -> e.g. `"23:59:59"` (String-encoded, HH:mm:ss)
     Time,
+
+    /// The following PlanetScale type IDs are mapped into DateTime:
+    /// - DATETIME (DATETIME) -> e.g. `"2023-01-01 23:59:59"` (String-encoded, yyyy-MM-dd HH:mm:ss)
+    /// - TIMESTAMP (TIMESTAMP) -> e.g. `"2023-01-01 23:59:59"` (String-encoded, yyyy-MM-dd HH:mm:ss)
+    DateTime,
```

Comment:
Note: TIMESTAMP is distinguished by DATETIME as it implies database-level UTC conversions that don't happen with DATETIME values. Quaint does not support this distinction. Can this lead to bugs we're not yet aware of?

Reply:
I'd open a new issue and do so in a follow-up PR. Does that work for you?